### ISSUE TITLE
How to release the memory after the call of UnloadModel

### ISSUE THREAD

**Issue by diann10101** [triton-inference-server/server#6836]
**Is your feature request related to a problem? Please describe.**
After called the Tritonserver_unloadmodel() and also stopserver(), the memory could not be released.
I followed the "Model Control Mode EXPLICIT", and LD_PRELOAD, but it didnot work.
I run the infer in Ubuntu 20.04.

**Describe the solution you'd like**
How to release the memory?

**Describe alternatives you've considered**
Any other API?

**Additional context**
Is it related to the build?
Is there any un-correct API usage?

**Comment by jbkyang-nvi** [comment]
Hello what version of Triton are you using? You can find it in the `TRTION_VERSION` part of the container. 
You seem to be using `tritonserver.so` and accessing the [in process API](https://github.com/triton-inference-server/core/blob/main/include/triton/core/tritonserver.h) or are using a server and accessing the server using a client?

If you want to just send a unload request on the client, you can call [UnloadModel()](https://github.com/triton-inference-server/client/blob/main/src/c%2B%2B/library/grpc_client.h#L289) if you are using grpc. 
If you are using the in-process API, then the direct call should unload the model.

It might be helpful to provide the full code snippet and steps to reproduce your issue too.

**Comment by diann10101** [comment]
Yes, the In-process-API and libtritonserver.so is used. The calling of UnloadModel() returned successfully. But the memory occupy by Linux Process is not releases.

**Comment by muyizi** [comment]
I have the same question.after I called UnloadModel(), but the memory is not released.
before unload:
<img width="670" alt="image-20240313" src="https://github.com/triton-inference-server/server/assets/13012815/fc018784-536c-4320-b509-a38d481871ea">


after unload:
<img width="678" alt="image-20240313 1" src="https://github.com/triton-inference-server/server/assets/13012815/f182110e-3cc1-4ac7-9b9d-d5cfd49e5819">

**Comment by RamitPahwa** [comment]
@muyizi / @diann10101 / @jbkyang-nvi any workarounds here, sending unload request, does not free up GPU memory

**Comment by RamitPahwa** [comment]
#7020
